WordPress/gutenberg
# Post navigation link: Add an option to only display arrows

## What?

Adds an option to hide the link text and only display the next and previous arrow icons.
The option uses a new block attribute, and is disabled by default.

Part of https://github.com/WordPress/gutenberg/issues/48443

## Why?

To give users and theme developers more design options.

## How?

When an arrow is selected, it displays a toggle with the text "Display arrow without text". 
Enabling this hides the text, places the icon inside the link, and also disables the "Display the title as a link" option.

On the front, "Next" and "Previous" are used as aria-labels to make sure there is a link text.

## Testing Instructions
Make sure that your WordPress installation has enough posts to navigate back and forward.
You can add the post navigation links directly in the posts, or, activate a block theme and add the
next and previous links to the **single** template.

1. To test the new setting, add the block and select either of the arrow icons.
2. When the icon is selected, confirm that you see a new toggle option with the text "Display arrow without text" below it.
3. Enable the option.
4. Confirm that the placeholder text disappears and only the icon shows.
5. The icon is clickable, but clicking on it should not open the target in the editor.
6. Save and view the block on the front. Confirm that only the icon shows, and that it is clickable and leads to the correct post.

7. Next, continue to test the other block options to test for any regressions.

Also test the color options. Even with the icon only option enabled, the block has `display:block` and the background color covers a much larger area than just the icon. There is room for improvement here.


### Testing Instructions for Keyboard


Place the block in either editor.
Navigate to the block settings sidebar.
Navigate past the variation selector and select the Settings tab (this tab is selected by default).
Navigate past the "Display the title as a link" option.
In the Arrow radio group option, select chevron or arrow.
Tab forwards to the next option "Display arrow without text", and check the box.
Please navigate back to the editor and select the block.
With the block selected, pressing the down arrow key or right arrow key should select the link.
If you are using a screen reader, the icon should be announced followed by the context: next or previous.
Save and go to the front. Locate the block. 
If you are using a screen reader, the link should have the link text Next or Previous.

## Screenshots or screencast 

https://user-images.githubusercontent.com/7422055/225238201-e8529e30-78b5-4ddc-a62a-91007fa0fd8e.mp4
